src/tools/generate/generate_chapters.js

```diff
@@ -12,7 +12,21 @@ converter.setFlavor('github');
 converter.setOption('simpleLineBreaks', false);
 converter.setOption('tablesHeaderId', false);
 
+const min_publish_date = new Date('2019-11-05T12:00');
+const sitemap_path = `templates/sitemap.xml`;
+
+var config = {};
+var chapters_config = [];
+var todo_chapters = [];
+
 const generate_chapters = async () => {
+
+  //Read in the Chapters to see which ones are still to do (needed for sitemap)
+  await read_config('2019');
+  chapters_config = config.outline.map(function(item) { return item["chapters"]; }).flat();
```

Comment:
Instead of mutating the module global `var`s, I find it's more sustainable to initialise them as `const` and then pass the reference into subsequent function calls? It makes the interdependencies between sections of functionality far more explicit.